custom_components/animated_scenes/animations.py

```
@@ -1142,50 +1101,82 @@ async def add_lights_to_animation(self, data: dict[str, Any]) -> None:
         if config.get(CONF_NAME, None) is not None:
             name: str = config.get(CONF_NAME)
         else:
-            name = self.hass.states.get(config.get(CONF_ANIMATED_SCENE_SWITCH)).attributes.get(
-                ATTR_FRIENDLY_NAME, config.get(CONF_ANIMATED_SCENE_SWITCH)
-            )
+            switch_entity_id = config.get(CONF_ANIMATED_SCENE_SWITCH)
+            registry_entry = er.async_get(self.hass).async_get(switch_entity_id)
+            if registry_entry is None:
+                _LOGGER.error("Animated Scene Switch %s was not found", switch_entity_id)
+                raise IntegrationError(f"Animated Scene Switch {switch_entity_id} was not found")
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)